### ISSUE TITLE
Regression: GRC fails to compile flow graphs with bus connections

### ISSUE THREAD

**Issue by argilo** [gnuradio/gnuradio#7059]
In GNU Radio 3.10.9.1, GRC fails to compile the following flowgraph:

![Screenshot from 2024-01-06 00-15-12](https://github.com/gnuradio/gnuradio/assets/583749/9ee60e29-028d-4f56-ae16-ce22bb43edd2)

The Null Source has two complex outputs, and the Null Sink has two complex inputs. Each block has "Bus Connections" set to `[[0,1]]`.

The exception is:

```
Traceback (most recent call last):
  File "/home/argilo/prefix_310/lib/python3.10/site-packages/gnuradio/grc/gui/Application.py", line 758, in _handle_action
    generator.write()
  File "/home/argilo/prefix_310/lib/python3.10/site-packages/gnuradio/grc/core/generator/top_block.py", line 86, in write
    for filename, data in self._build_python_code_from_template():
  File "/home/argilo/prefix_310/lib/python3.10/site-packages/gnuradio/grc/core/generator/top_block.py", line 135, in _build_python_code_from_template
    connections=self._connections(),
  File "/home/argilo/prefix_310/lib/python3.10/site-packages/gnuradio/grc/core/generator/top_block.py", line 347, in _connections
    connection = fg.parent_platform.Connection(
  File "/home/argilo/prefix_310/lib/python3.10/site-packages/gnuradio/grc/gui/canvas/connection.py", line 39, in __init__
    super(self.__class__, self).__init__(*args, **kwargs)
  File "/home/argilo/prefix_310/lib/python3.10/site-packages/gnuradio/grc/core/Connection.py", line 56, in __init__
    param_factory = self.parent_platform.make_param
  File "/home/argilo/prefix_310/lib/python3.10/site-packages/gnuradio/grc/core/utils/descriptors/_lazy.py", line 19, in __get__
    value = self.func(instance)
  File "/home/argilo/prefix_310/lib/python3.10/site-packages/gnuradio/grc/core/base.py", line 108, in parent_platform
    return self.get_parent_by_type(Platform)
  File "/home/argilo/prefix_310/lib/python3.10/site-packages/gnuradio/grc/core/base.py", line 103, in get_parent_by_type
    return parent.get_parent_by_type(cls)
AttributeError: 'TopBlockGenerator' object has no attribute 'get_parent_by_type'
```

A bisect shows that the problem began with https://github.com/gnuradio/gnuradio/pull/6889.

**Comment by argilo** [comment]
Strangely, if bus sizes are mismatched, then the flow graph compiles without issue.

**Comment by willcode** [comment]
Similar problem with virtual sink/source was fixed in #7016

**Comment by mbr0wn** [comment]
I think I know how to fix this, but I can't replicate the original issue. I'm having issues with all examples that have bus ports, though.
